src/mmap_unix.rs

```diff
@@ -74,6 +74,152 @@ impl error::Error for Error {}
 
 pub type Result<T> = result::Result<T, Error>;
 
+/// A factory struct to build `MmapRegion` objects.
+pub struct MmapRegionBuilder<B = ()> {
+    size: usize,
+    prot: i32,
+    flags: i32,
+    file_offset: Option<FileOffset>,
+    raw_ptr: Option<*mut u8>,
+    hugetlbfs: Option<bool>,
+    bitmap: Option<B>,
+}
+
+impl<B: Bitmap + NewBitmap> MmapRegionBuilder<B> {
```

Comment:
Isn't `NewBitmap` enough here?

Reply:
The with_bitmap() method needs "B: Bitmap" bound.

Reply:
Do we have to add it explicitly if NewBitmap is a Bitmap itself?

Reply:
yeah, not realized `NewBitmap: Bitmap`.